src/components/i-calendar.vue

```diff
@@ -0,0 +1,537 @@
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import dayjs from 'dayjs';
+
+import IcAngleCircle from '@/icons/ic-angle-circle.vue';
+
+export default {
+  name: 'ICalendar',
+  components: {
+    IcAngleCircle,
+  },
+  props: {
+    value: {
+      type: [Date, Array],
+      default: () => null,
+    },
+    initialDate: {
+      type: Date,
+      default: undefined,
+    },
+    disabledPreviousMonth: Boolean,
+    disabledNextMonth: Boolean,
+    isThereNewUpdate: Boolean,
+  },
+  data() {
+    return {
+      activeDate: undefined,
+      activeMonth: null,
+      activeYear: null,
+      selectedDate: [],
+      activeMonthPicker: undefined,
+      activeYearPicker: undefined,
+      showMonthYearTable: false,
+    };
+  },
+  computed: {
+    listDays() {
+      return Array.from(Array(7), (v, i) =>
+        dayjs()
+          .day(i + 0)
+          .format('ddd'),
+      );
+    },
+    listMonth() {
+      const x = Array.from(Array(12), (v, i) => ({
+        monthDisplay: dayjs(this.activeYearPicker)
+          .month(i + 0)
+          .format('MMM'),
+        monthValue: dayjs(this.activeYearPicker)
+          .month(i + 0)
+          .startOf('month'),
+      }));
+
+      return x;
+    },
+    activeDateString() {
+      return this.activeDate ? this.activeDate.toString() : undefined;
+    },
+    activeYearPickerString() {
+      return dayjs(this.activeYearPicker.toString()).format('YYYY');
+    },
+    previousPicker() {
+      const display = [];
+
+      const previous = dayjs(this.activeDateString).date(0);
+      const current = dayjs(this.activeDateString).date(0);
+      for (let i = 0; i <= current.day(); i += 1) {
+        display.push(previous.subtract(i, 'day'));
+      }
+      return display.sort((a, b) => a.day() - b.day());
+    },
+    currentPicker() {
+      return Array.from(Array(dayjs(this.activeDateString).daysInMonth()), (v, i) => {
+        const date = dayjs(this.activeDateString).date(++i);
+        return { date };
+      });
+    },
+    nextPicker() {
+      const display = [];
+      const previous = this.previousPicker.length;
+      const current = dayjs(this.activeDateString).daysInMonth();
+      for (let i = 1; i <= 42 - (previous + current); i += 1) {
+        display.push(dayjs(this.activeDateString).date(i).add(1, 'month'));
+      }
+      return display;
+    },
+  },
+  watch: {
+    activeDate: {
+      immediate: true,
+      handler(val) {
+        if (val) {
+          this.activeMonth = dayjs(val.toString()).format('MMM');
+          this.activeYear = dayjs(val.toString()).format('YYYY');
+        }
+      },
+    },
+    value: {
+      handler(val) {
+        const valueArray = Array.isArray(val) ? val : [val];
+
+        const isValueSame = this.checkSame(valueArray, this.selectedDate);
+        if (!isValueSame) {
+          this.selectedDate = valueArray.filter((date) => !!date).map((date) => dayjs(date.toString()).toDate());
+        }
+      },
+    },
```

Comment:
create selectedDate watch (you can use i-datepicker as reference)
add :
this.$emit('onSelectDate', 'add the date here');
this.$emit('input'); (this one to update v-model at parent)

Reply:
ok